Backend: Keybindings

rueblimaster opened the pull request:
## Dependencies
-  #3871

## What
Describe what this pull request does, including technical details, screenshots, links to discord, etc.

<details>
<summary>Images</summary>

<!-- drop images here -->

</details>

## Changelog Technical Details
+ Added Keybinding to simplify keybindings. - rueblimaster

rueblimaster committed:
Create Keybinding.kt

rueblimaster committed:
improve keybindings

rueblimaster committed:
made conditions nullable

rueblimaster committed:
massively improve keybindingType etc.

rueblimaster committed:
small cleanup / improvements

rueblimaster committed:
add questionable command

rueblimaster committed:
better names

rueblimaster committed:
use fancy new HandleEvent annotation

rueblimaster committed:
Revert "use fancy new HandleEvent annotation"

This reverts commit 40dc592fb06dd2877d70f1ffb9ac9e4fbe993258.

rueblimaster committed:
add back DebugDataCollect

rueblimaster committed:
fix wrong keybindingType

rueblimaster committed:
some detekt

rueblimaster committed:
use fancy new HandleEvent annotation

rueblimaster committed:
remove activeKeybindings

as the performance gain is only minor and not worth it

rueblimaster committed:
improve DebugDataCollect

rueblimaster committed:
remove keybindingType caching

rueblimaster committed:
implement in FocusMode

rueblimaster committed:
implement PestFinder

rueblimaster committed:
implement in GardenWarpCommands

rueblimaster committed:
implement in FishingTimer

rueblimaster committed:
detekt

rueblimaster committed:
add option requireSeparateTaps

rueblimaster committed:
detekt

hannibal002 commented:
I'd rather see an key up event that also includes the "last key up of same key duration" as a parameter

I see you copied the approach for the inventory render helper here. But I'm not sure I like this solution for keybinds

rueblimaster committed:
Merge branch 'beta' into KeyBindUtils

rueblimaster committed:
names?

rueblimaster commented:
only going to be done after
- https://github.com/hannibal002/SkyHanni/pull/3871

rueblimaster committed:
Merge branch 'beta' into KeyBindUtils

rueblimaster committed:
KeyStateManager (#7)

* it, I hope

* detekt

* final touches

* detekt

rueblimaster committed:
improve KeyHeldEvent posts

rueblimaster committed:
KeyHeldEvent also with mouse

rueblimaster committed:
use KeyEvents

rueblimaster committed:
imports

rueblimaster committed:
detekt

rueblimaster committed:
baselined

rueblimaster committed:
Revert "baselined"

This reverts commit c4232fb9c1bfc19da28605987ca7653ef69d9e3d.

rueblimaster committed:
baselined, but correctly

rueblimaster committed:
Update baseline.xml

rueblimaster committed:
exclude from 1.21

rueblimaster committed:
Create baseline.xml

rueblimaster committed:
Merge branch 'beta' into KeyBindUtils

rueblimaster committed:
merge from KeyStateManager

rueblimaster committed:
formatting

rueblimaster committed:
Merge branch 'beta' into KeyBindUtils

rueblimaster committed:
fix bad merge

rueblimaster committed:
update

rueblimaster committed:
Merge branch 'beta' into KeyBindUtils

rueblimaster committed:
Merge branch 'beta' into KeyBindUtils

CalMWolfs commented:
Have not looked through this pr yet but I would like this to not be merged before we have a working 1.21 version. As i dont want to debug completely new code after already changing a bunch of stuff to get key events to work on 1.21. Merging this will just mean there is more uncertainty as to what broke key events and will slow down fixing it as well. 
I will take it off soon but it can probably be re-added in a day or two

rueblimaster committed:
Merge branch 'beta' into KeyBindUtils

rueblimaster committed:
Remove unused KeybindHelper import

Deleted the import of KeybindHelper from moulconfig as it was not being used in KeyboardManager.kt.

rueblimaster commented:
> Have not looked through this pr yet but I would like this to not be merged before we have a working 1.21 version. As i dont want to debug completely new code after already changing a bunch of stuff to get key events to work on 1.21. Merging this will just mean there is more uncertainty as to what broke key events and will slow down fixing it as well. I will take it off soon but it can probably be re-added in a day or two

This doesn't change the key events in any way or form, only the dependency does

rueblimaster committed:
Merge branch 'beta' into KeyBindUtils

rueblimaster committed:
Merge branch 'beta' into KeyBindUtils

rueblimaster committed:
Merge branch 'beta' into KeyBindUtils

rueblimaster committed:
Refactor key event handling and remove KeyUpEvent

Replaces KeyDownEvent and KeyUpEvent with KeyPressEvent, KeyReleaseEvent, and KeyHeldEvent for improved key event granularity. Updates event imports and usages in overlays and mixins, and removes the obsolete KeyUpEvent class.

rueblimaster committed:
Update custom dictionary entries

Reorganized existing entries in default_user.xml to improve spellchecking for project-specific terminology.

rueblimaster committed:
Merge branch 'beta' into KeyBindUtils

rueblimaster committed:
Merge branch 'beta' into KeyBindUtils

rueblimaster committed:
Merge branch 'beta' into KeyBindUtils
